crates/burn-jit/src/tensor/layout.rs

```diff
@@ -0,0 +1,121 @@
+#[derive(PartialEq, Eq, Debug)]
+pub(crate) enum MemoryLayout {
+    /// Memory is wholly contiguous, with row major layout
+    Contiguous,
+    /// Permutations happened, but may not impact some kernels
+    MildlyPermuted {
+        /// Last two dims are inverted
+        transposed: bool,
+        /// Some permutations exist in batch dimensions
+        batch_swap: bool,
```

Comment:
We assume the the batch dimensions are all except the two lasts, which is not always the case. We should put a comment here